### ISSUE TITLE
正样本是随机取一个吗？

### ISSUE THREAD

**Issue by kennyLSN** [FlagOpen/FlagEmbedding#857]
def __getitem__(self, item) -> List[BatchEncoding]:
        query = self.dataset[item]['query']
        pos = random.choice(self.dataset[item]['pos'])
        if len(self.dataset[item]['neg']) < self.args.train_group_size - 1:
            num = math.ceil((self.args.train_group_size - 1) / len(self.dataset[item]['neg']))
            negs = random.sample(self.dataset[item]['neg'] * num, self.args.train_group_size - 1)
        else:
            negs = random.sample(self.dataset[item]['neg'], self.args.train_group_size - 1)

        batch_data = []
        batch_data.append(self.create_one_example(query, pos))
        for neg in negs:
            batch_data.append(self.create_one_example(query, neg))

        return batch_data

**Comment by staoxiao** [comment]
Yes, we will randomly select a positive from the positive list when generate pair data for training.

**Comment by kennyLSN** [comment]
> Yes, we will randomly select a positive from the positive list when generate pair data for training.

这样是不是没办法学到更多正样本关系

**Comment by staoxiao** [comment]
This code will sample a different positive for each step, so it can utilize all positive texts when training multiple epochs.
